source/ingress.go

```diff
@@ -82,6 +85,19 @@ func (sc *ingressSource) Endpoints() ([]*endpoint.Endpoint, error) {
 			continue
 		}
 
+		// Skip sources not matching annotation filter
+		annotation, pattern := splitAnnotationFilter(sc.annotationFilter)
+		string := ing.Annotations[annotation]
+		matched, err := regexp.MatchString(pattern, string)
+		if err != nil {
+			return nil, err
+		}
+		if len(pattern) > 0 && !matched {
+			log.Debugf("Skipping ingress %s/%s because '%s' annotation does not match pattern, found: %s, expected: %s",
+				ing.Namespace, ing.Name, annotation, string, pattern)
+			continue
+		}
```

Comment:
I recommend adopting this pattern: https://github.com/linki/chaoskube/blob/v0.6.0/chaoskube/chaoskube.go#L180-L199

With this you get all the matching functionality from Kubernetes itself for free, including matching mutliple labels, checking for existence (`--filter=foo`) as well as exclusion (`--filter=ingress.class!=nginx`).